src/com/ableton/Pyenv.groovy

```diff
@@ -43,11 +46,23 @@ class Pyenv implements Serializable {
       script.error 'This method is not supported on Windows'
     }
 
+    if (!versionSupported(pythonVersion)) {
+      script.withEnv(["PYENV_ROOT=${pyenvRoot}"]) {
+        String pyenvVersion = script.sh(
+          label: 'Get Pyenv version on the node',
+          returnStdout: true,
+          script: "${pyenvRoot}/bin/pyenv --version",
+        ).trim()
+        script.error "The installed version of Pyenv (${pyenvVersion}) does not " +
+          "support Python version ${pythonVersion}"
+      }
+    }
+
     VirtualEnv venv = new VirtualEnv(script, randomSeed)
-    int result = venv.script.sh(
-      label: "Install Python version ${pythonVersion} with pyenv",
-      returnStatus: true,
-      script: """
+    script.retry(INSTALLATION_RETRIES) {
+      venv.script.sh(
+        label: "Install Python version ${pythonVersion} with pyenv",
+        script: """
```

Comment:
I think the indent level is off for the below lines, sorry for not noticing that before.

Reply:
No worries, see 5a5876fb3a09e2f0ba02dfc9f64487f1633198da.